templates/distribution/manifests/dr/kustomization.yaml.tpl

```
@@ -41,13 +61,65 @@ patchesStrategicMerge:
   - patches/velero-schedule-manifests.yml
   - patches/velero-schedule-full.yml
 {{- end }}
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if ne .spec.distribution.modules.dr.etcdBackup.type "none" }}
+  - patches/etcd-backup-schedule.yml
+{{- end }}
+{{- end }}
+
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if ne .spec.distribution.modules.dr.etcdBackup.type "none" }}
+configMapGenerator:
+{{- if eq .spec.distribution.modules.dr.etcdBackup.type "all" "s3" }}
+  - name: etcd-backup-s3-config
+    behavior: replace
+    literals:
+      - target={{ print "minio:" .spec.distribution.modules.dr.etcdBackup.s3.bucketName }}
+      - retention={{ .spec.distribution.modules.dr.etcdBackup.s3.retentionTime }}
+    {{- if eq (index .spec.distribution.modules.dr.etcdBackup "backupPrefix") nil }}
+      - backup-prefix={{ print .metadata.name "-" }}
+    {{- else}}
+      - backup-prefix={{ .spec.distribution.modules.dr.etcdBackup.backupPrefix }}
+    {{- end}}
```

You can just check that index is true, that makes it a little more legible, 
I HAVEN'T TESTED IT, but I'm pretty sure this works:

```suggestion
    {{- if index .spec.distribution.modules.dr.etcdBackup "backupPrefix" }}
      - backup-prefix={{ .spec.distribution.modules.dr.etcdBackup.backupPrefix }}
    {{- else}}
      - backup-prefix={{ print .metadata.name "-" }}
    {{- end}}
```

---

Yes, I specifically checked for `nil` because otherwise Go's template engine considers a nil value (i.e. a non existing field) the same as the zero value (falsy-values). 

The rationale here is if `backupPrefix`'s left unset (i.e. nil value) we apply the cluster's name, if empty we want an empty prefix (i.e. only the date + .etcdb).

---

makes sense

---

I can add a comment to better explain the rationale!

---

that would be great